src/nvim/os/input.c

```diff
@@ -316,7 +317,7 @@ static void read_cb(RStream *rstream, void *data, bool at_eof)
 
 static void process_interrupts(void)
 {
-  if (mapped_ctrl_c) {
+  if ((mapped_ctrl_c | curbuf->b_mapped_ctrl_c) & get_real_state()) {
```

Comment:
I would appreciate if someone familiar with the character input handling in Neovim (@tarruda?) could check if this is the proper place to put this check (read `curbuf->b_mapped_ctrl_c` and `get_real_state()`).